### ISSUE TITLE
rate limit scans per target

### ISSUE THREAD

**Issue by jvehent** [mozilla/tls-observatory#87]
The scanners should refuse to scan a given target more than once every X minutes (let's say 3, for now). This could be done by dropping scans using a query like
```sql
SELECT id
FROM scans
WHERE target=$1
AND timestamp < NOW() - INTERVAL '3 minutes';
```
A dropped scan would get picked up to run in a 2nd attempt, so no need to implement a new queueing mechanism.

**Comment by 0xDiBa** [comment]
As I understand it you mean that scanners will not pick up scans depending on the above query.
I agree that this rate limits current scans but the scans are going to occur anyway in intervals every 3 mins.

Would it be right to implement this mechanism in the API frontend so that scans are not even queued multiple times? Maybe this can be implemented along with #84, although the logic and purpose of the two is different.

**Comment by jvehent** [comment]
It would make sense to implement in the API, and return a `429 Too Many Requests` to indicate that the scan was rejected.

**Comment by 0xdiba** [comment]
I am implementing this right now, do you think in the above case, along with the 429 status code, we should return the latest scan id or an "error" describing the situation more?

The scan id might prove useful but I think it would be a little bit messy and would need documenting (#90).

**Comment by jvehent** [comment]
There may be 2 cases here:
1. `scan?target=***&force=false` then the most recent scan for the target should be returned with `HTTP 200`
2. `scan?target=***&force=true` then the scan is refused and a `HTTP 429` with an error is returned

**Comment by jvehent** [comment]
fixed by #94
